File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/dashboards/widgets/PromotionFrequencyChartWidget.kt
package net.nemerosa.ontrack.model.dashboards.widgets

import org.springframework.stereotype.Component

@Component
class PromotionFrequencyChartWidget: AbstractWidget<PromotionChartWidgetConfig>(
    key = "home/PromotionFrequencyChart",
    name = "Promotion frequency",
    description = "Chart displaying the number of times a promotion occurs during the selected period.",
    defaultConfig = PromotionChartWidgetConfig(project = null, branch = null, promotionLevel = null),
    preferredHeight = 6,
)

File: ontrack-web-core/components/widgets/home/PromotionFrequencyChartWidget.js
import PromotionLevelFrequencyChart from "@components/promotionLevels/PromotionLevelFrequencyChart";
import PromotionChartWidget from "@components/widgets/home/PromotionChartWidget";

export default function PromotionFrequencyChartWidget(props) {
    return (
        <PromotionChartWidget
            prefix="Frequency of"
            chart={PromotionLevelFrequencyChart}
            {...props}
        />
    )
}
